src/main/java/com/google/codeu/servlets/SingleConversationTopicServlet.java

```diff
@@ -15,6 +22,28 @@ public void init() {
         datastore = new Datastore();
     }
 
-    /** Responds with a JSON representation of the {@link ConversationTopic} data for the specific chatroom */
+    /**
+     * Responds with a JSON representation of the {@link Messages} data for
+     * the specific chatroom
+     */
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        System.out.println("SingleConversationTopicServlet - getting messages for a specificconversation topic");
+
+        response.setContentType("application/json");
+
+        String conversationTopicId = request.getParameter("id");
 
+        if (conversationTopicId == null || conversationTopicId.equals("")) {
+			// Request is invalid, return empty array
+			response.getWriter().println("[]");
+			return;// stop
+		}
+
+        List<Message> messages = datastore.getConversationTopicMessages(conversationTopicId);
```

Comment:
You can query the conversation topics entity for the the title/name here and send it down as part of the JSON